libnavui-androidauto/src/main/java/com/mapbox/androidauto/notification/CarNotificationInterceptor.kt

```diff
@@ -0,0 +1,135 @@
+package com.mapbox.androidauto.notification
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import androidx.car.app.model.CarColor
+import androidx.car.app.notification.CarAppExtender
+import androidx.car.app.notification.CarPendingIntent
+import androidx.core.app.NotificationCompat
+import com.mapbox.androidauto.R
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.TimeFormat
+import com.mapbox.navigation.base.formatter.DistanceFormatter
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.formatter.MapboxDistanceFormatter
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.core.trip.session.NavigationSessionState
+import com.mapbox.navigation.core.trip.session.NavigationSessionStateObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+
+/**
+ * Register this observer using [MapboxNavigationApp.registerObserver]. As long as it is
+ * registered, the trip notification will be shown in the car.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+class CarNotificationInterceptor internal constructor(
+    private val context: Context,
+    private val pendingOpenIntent: PendingIntent?,
+    private val setupNotification: NotificationCompat.Builder.() -> Unit,
+    private val setupCarAppExtender: CarAppExtender.Builder.() -> Unit,
```

Comment:
I think we should remove these for now. It's not clear they're needed without a example use case or tests.

Reply:
I thought about giving the option to set high importance, so that a heads-up notification is shown. Such notifications distract to the driver and they should be shown only when "there is a significant navigation turn event". By default the notification is shown only in the rail widget. But I don't mind removing that option for now.

Reply:
Here is what a heads-up notification looks like:
<img width="912" alt="Screenshot 2022-05-17 at 00 10 23" src="https://user-images.githubusercontent.com/2395284/168683185-704fc295-defd-4565-81cf-583136808c9c.png">

Reply:
> I thought about giving the option to set high importance, so that a heads-up notification is shown

yeah, considering that there is a `TripNotificationInterceptor`. Maybe we could use this to popocate a `CarTripNotificationInterceptor`

The `CarTripNotificationInterceptor` allows you to provide further customizations like importants 🤔 . But something like that would be nice to have a explicit class for. That way it is easy to reference and document.